Sources/Register.php

```diff
@@ -50,6 +50,10 @@ function Register($reg_errors = array())
 	$context['require_agreement'] = !empty($modSettings['requireAgreement']);
 	$context['registration_passed_agreement'] = !empty($_SESSION['registration_agreed']);
 	$context['show_coppa'] = !empty($modSettings['coppaAge']);
+	
+	// Do we nee them to agree to the policy agreement, secondly?
+	$context['require_policy'] = !empty($modSettings['enable_policy_function']);
```

Comment:
Here would be another good place to add more logic to make sure that all the necessary variables are defined before requiring the policy.